app/src/main/java/com/zulip/android/filters/NarrowFilterSearch.java

```diff
@@ -48,7 +48,8 @@ public NarrowFilterSearch(String query) {
 
     @Override
     public boolean matches(Message msg) {
-        return msg.getContent().toLowerCase().contains(query.toLowerCase());
+        //api matching for us
```

Comment:
Yes, it appears that the filters were attempting to filter already filtered results. When I removed this line, the results were already filtered, and it makes sense that the backend would do the filtering regardless